Multi IP environment fix

kotso opened the pull request.

kotso committed:
Multi IP environment fix

sandrodz commented:
Hi,

Comment suggests "IP address of customer..." this is confusing. As we are setting client IP addresses here https://github.com/wearede/tbcpay-php/blob/master/src/WeAreDe/TbcPay/TbcPayProcessor.php#L184

While I understand what you meant with customer, it was confusing at first glance. 

From: https://curl.haxx.se/libcurl/c/CURLOPT_INTERFACE.html
Sets the interface name to use as outgoing network interface. The name can be an interface name, an IP address, or a host name.

Maybe use desc from curl docs and reword comment for clarity.

kotso committed:
Remove duplicate src_ip variable

* remove duplicate src_ip variable (client_ip_addr already defines required value)
* if client_ip_addr is set then set it as interface source ip for curl

In multi IP environment when website IP is not default server IP address website ip and default ip curl initiates connection defers, this causes gateway to reject requests as it appears to be originated from wrong ip.

kotso commented:
Updated code, remove src_ip and added client_ip_address to be curl interface also.
